### ISSUE TITLE
Questie multiple errors, Tracker not working

### ISSUE THREAD

**Issue by Schattenthron** [AeroScripts/QuestieDev#218]
Hello,

I'm using the latest release version of Questie (v3.0).
Playing on a German Classic Server 1.12.1.

I'm using AddOn !ImprovedErrorFrame (german localization).
My attempt at a translation will be written inside []-brackets.

The following errors occur after Wow-login:
1. Fehler [error]: **attempt to index global 'QuestieConfig' (a nil value)** ; AddOn: !Questie ; Datei [file]: **QuestieTracker.lua** ; Zeile [line]: **884** ; Fehler Nummer [error count?]: Unendlich [infinite]
2. Fehler [error]: **attempt to index field '?' (a nil value)** ; AddOn: !Questie ; Datei [file]: **QuestieQuest.lua** ; Zeile [line]: **104** ; Fehler Nummer [error count?]: Unendlich [infinite]

Quest tracker and arrow do not show up ingame and everytime I try to manually track a quest I will get the error:
- Fehler [error]: **attempt to index field '?' (a nil value)** ; AddOn: !Questie ; Datei [file]: **QuestieQuest.lua** ; Zeile [line]: **59** ; Fehler Nummer [error count?]: 1

Quests show up on the mini map and world map.
I'm using AddOn MetaMap and Questie is the only Questhelper-AddOn installed.

**Comment by Dyaxler** [comment]
This has been hot-fixed and the ZIP file has been updated. Please give it a try.

https://github.com/AeroScripts/QuestieDev/releases/latest

**Comment by Schattenthron** [comment]
Hello Dyaxler,

I have installed your hot-fixed version.

The last time I forgot to mention that I had more errors caused by the Lib Babble-Zone-2.2.lua.
After updating the Babble-Zone-2.2.lua in the !Questie -> Libs -> Babble-Zone-2.2 directory with a newer one, the following errors disappear:

1. Fehler [error]: **AceLocale(Babble-Zone-2.2): Can only accept true as a value on the base locale. "enUS" is the base locale, "deDE" is not.** ; Datei [file]: **...ns\!Questie\Libs\Babble-Zone-2.2\Babble-Zone-2.2.lua** ; Zeile [line]: **177** ; Fehler Nummer [error count?]: 1
2. Fehler [error]: **Tourist-2.0 requires Babble-Zone-2.2.** ; AddOn: !Questie ; Datei [file]: **Tourist-2.0.lua** ; Zeile [line]: **18** ; Fehler Nummer [error count?]: 1

The newer Babble-2.2 Libs I used can be found here:
http://www.wowace.com/addons/babble-2-2/files/195-babble-2-2-r19548/

Babble-2.2.toc is ## Interface: 20000, but the Babble-Zone-2.2.lua seems to work. ;D

My setup for testing:

- Playing on a German classic server 1.12.1.
- AddOns used: !ImprovedErrorFrame (german localization), Questie v3.0
- All other AddOns disabled
- Updated Babble-Zone-2.2.lua

This one is tricky because I don't know how a fully functional Questie has to look like.
I can only compare the feature description and screenshots from the Nostalrius forum with my results ingame.

What seems to work:

- This time no errors occur after Wow-login.
- Quests show up on the mini map and world map (only in English).
- I can manually complete a quest by shift-clicking it, possible both on minimap and world map.
- On world map I can "Toggle Questie" to hide/show the quests.

What seems broken:

- Quest tracker and arrow do not show up, I have only access to the vanilla quest log from the menu
- If I try to manually track (shift-click) a quest fom vanilla quest log nothing happens

Maybe more of the old Libs in the !Questie directory are broken?
I don't know.

Greetings

**Comment by Dyaxler** [comment]
I didn't update or touch the ace libraries in the Vanilla client only in the TBC client... maybe you found a new bug? I'll look into the Babble-zone bug but I'm not comfortable replacing it with a version from TBC.

Question about your QuestLog... are the titles in English or in German? I suspect they are in German which was causing the original error to begin with. The fix I applied looks for a valid quest. If it's not found the rest of the function is exited silently and no data gets stored in your quest tracker; hence the reason it's not showing up in your tracker.

The manual (shift-click) finds the quest hash by the tool-tips quest title, not the quest log, so that's why that feature works.

Does the German client "translate" English into German with a special global strings file or is it hard coded with German exclusively? Another way of looking at my question, can you make your quest log display the quests in English in the German WoW client? If so then I can look at possibly coding around this issue. Example, "if questie can't find the quest using the quest logs quest title, fall back and bypass the wow client function that is translating the quest title from English to German and look again. I did something similar with Russian players using a translator mod called wowRu - I simply blocked two global strings from being translated within Questie allowing the mod to continue finding quests properly.

**Comment by Dyaxler** [comment]
... another thought I just had about Babble-Zone. It's possible that we're looking at purely a localization issue. The older version of Babble-Zone wasn't handling the German client correctly where as the newer version was.

Once I hotfix, again if possible, your issue with finding the quests... the newer Babble-Zone file you're using might not work anymore. Or it might. :)

**Comment by Schattenthron** [comment]
Hello Dyaxler,

_"Question about your QuestLog... are the titles in English or in German?"_
I'm indeed using a German Wow client 1.12.1.
So all quests titles and text in the vanilla quest log are in German.

_"Does the German client "translate" English into German with a special global strings file or is it hard coded with German exclusively?"_
Sry I don't really know. Acoording to this thread (https://community.mmobase.de/talk-und-allgemeines/1913-wow-auf-englisch.html), it seems like all the names of the NPCs, Gameobjects, Items, etc. in the database of the German Wow server are in German (http://www.classic-wow.org/).
Quick summary: A user asks if it's possible to play the game in English and then a developer of the server answers that the names of the database are in German and that it's possible to use an English client but the results will be strange because the names ingame will be a mixture of English and German.

_"Another way of looking at my question, can you make your quest log display the quests in English in the German WoW client?"_
Unfortunately there is no option to change the language ingame, possibly with an AddOn. But I will try to download and install the English Wow client and look if the quest titles are in English.

Yes I also think it's a localization issue.

**Comment by Dyaxler** [comment]
I just spoke with a German friend. Yes the German servers database is all in German. Unfortunately, Questie isn't officially supported on those servers. Even if you downloaded an English client the quests being presented from the server and in your quest log will still be in German. Questie has no way of translating those quests into English.

I'm investigating an idea of localizing Questie but I can't promise anything.

**Comment by Schattenthron** [comment]
That's a pity. I would really like to use Questie, looks like a nice questhelper.

Maybe I can help somehow, translating or testing?
Unfortunately I have no programming experience.
(It was many years ago when I programmed some basic Java stuff at university) ;D

There is this German fellow Eric Mauser who developed the AddOns ShaguQuest and ShaguDB.
According to his website (http://shaguaddons.ericmauser.de/shaguquest/) his database is based on the latest Mangos Zero Database, as well as German Mangos DataBase for the deDE versions. He offers a German version of his AddOn and even mentions Questie.

So maybe a collaboration is possible and Questie + ShaguQuest can be merged into some **Uber**-questhelper AddOn for vanilla Wow. ;D

Greetings

**Comment by Dyaxler** [comment]
His database contains unique data specific for Shagu... it's not compatible. :(

One of the original authors for Questie is working on something that might bring Questie up to speed and allow me to import a localized German DB... but again I can't make any promises. I've asked them to update a tool they used to build the original DB for Questie so that I could use it to build a TBC and Wotlk DB for the QuestieTBC and QuestieWrath clients I'm working on. Its my hope that I will also have the ability to build DB's in other languages.

**Comment by chaosraven** [comment]
Hi Dyaxler,

I'm also working on a german Client, but playing on kronos (english server). I got the error already posted from Schattenthron:
"AceLocale(Babble-Zone-2.2): Can only accept true as a value on the base locale. "enUS" is the base locale, "deDE" is not."
Next thing I have noticed, is that my BigWigs doesn't start up.
These two problems got solved with the updated Babble-version listed above

The questtracker isn't showing up by standard. I've to type "/questie cleartracker" first, after this step I am be able to track my quests, but the quest-tracker stucks somewhere at the center of my screen. I haven't found a way to drag&drop it to the right place and after I have used "/questie header" it got anchored at the top-left corner, still unmoveable.

It also seems like the database isn't full already, cause I'm missing a lot quests on the world and minimap.

Last but not least, where is my navigation-arrow? :-(

Kind regards chaosraven

**Comment by Vanholtz** [comment]
Hello Dyaxler,

Yesterday afternoon (15/05/2016) I've installed Questie v3.0 for the first time and I faced the same issue than chaosraven concerning AceLocale(Babble-Zone-2.2). The only difference is that I'm using a French WoW client 1.12.1.

Here is the extraction of the file FrameXML.log:
5/16 00:59:29.686  Loading add-on !Questie
5/16 00:59:29.686  ** Loading table of contents Interface\AddOns\!Questie\!Questie.toc
5/16 00:59:29.686  ...ns\!Questie\Libs\Babble-Zone-2.2\Babble-Zone-2.2.lua:322: AceLocale(Babble-Zone-2.2): Can only accept true as a value on the base locale. "enUS" is the base locale, "frFR" is not.
5/16 00:59:29.686  ...ace\AddOns\!Questie\Libs\Tourist-2.0\Tourist-2.0.lua:18: Tourist-2.0 requires Babble-Zone-2.2.
5/16 00:59:29.686  Error loading Interface\AddOns\!Questie\Database\hash.lua

I saw you released a few minutes ago a new release v3.1 but of course the result is unfortunately the same, we still got the issue.

It would be great of we could find someone that could fix this bug at the game start.

See you,
Vanholtz

**Comment by Dyaxler** [comment]
French client... I assume a server that is also using a French database? Questie is currently only compatible with English.

> 5/16 00:59:29.686 ...ns!Questie\Libs\Babble-Zone-2.2\Babble-Zone-2.2.lua:322: AceLocale(Babble-Zone-2.2): Can only accept true as a value on the base locale. "enUS" is the base locale, "frFR" is not.

**Comment by Vanholtz** [comment]
Of course the server is in French, too.
Where is the problem to make this addon compatible with German and French?
No localization is planned (like almost all other addons do)?

**Comment by Dyaxler** [comment]
Since the server doesn't provide "QuestID's" like the later versions of World of Warcraft does, Questie relies on finding quest data by the quest name. It looks at an internal database for this info by matching the quest title. This internal database is currently only in English. So if a server says that X quest's name is (if in French or German) then Questie won't find X quest in it's database because it's expecting English to match an English quest title.

Like I've already mentioned in several posts in the past, and again I can't make any promises but we're working on a tool that will build a French or German database to make it compatible. The person working on the tool is very busy so I don't have an ETA on when or if it will happen. This same tool can hopefully be used to build a TBC or a WotLK database too.

**Comment by Vanholtz** [comment]
Ok, I'll be patient then :-)
Thanks!

**Comment by Dyaxler** [comment]
You're very welcome. :)

**Comment by Maximos118** [comment]
Hi! I also have this issue! I'm English. I'm using Client 1.12.1. I'm trying to use Questie on Elysium as it looks really helpful but when I load in... all i get are mini-map and full map symbols. Quest tracker and Arrow isn't showing at all! I'd really like this functionality! I'm using Quiestie 3.61.

I've tried previous versions of Questie, the Dev versions of Questie, I've tried reinstalling wow, I've tried removing all other add-ons.

Any ideas?!
